agent/__init__.py

```diff
@@ -194,98 +199,95 @@ def generate_cert(device_id):
-        requester_name, req_type, response.content))
+
+    if log_on_ok or not response.ok:
+        logger.debug("{} :: [RECEIVED] {} {}: {}".format(caller, requester, req_type, response.status_code))
+        logger.debug("{} :: [RECEIVED] {} {}: {}".format(caller, requester, req_type, response.content))
 
 
-def mtls_request(method, url, debug=False, dev=False, requester_name=None, debug_on_ok=False,
-                 return_exception=False, **kwargs):
+def mtls_request(method, url, dev=False, requester_name=None, log_on_ok=False, return_exception=False, **kwargs):
     """
     MTLS  Request.request wrapper function.
     :param method = 'get,'put,'post','delete','patch','head','options'
     :param url: request url string (without endpoint)
-    :param debug: if true then log error messages
     :param dev: if true use dev endpoint and dev headers
     :param requester_name: displayed requester id for error messages
-    :param debug_on_ok: if true with debug then log successful response status/content too
-    :param return_exception if true, then returns tuple ( response, None ) or (None, RequestException)
+    :param log_on_ok: if true then log debug message even if response.ok
+    :param return_exception: if true, then returns tuple ( response, None ) or (None, RequestException)
     :return: response or None (if there was exception raised), or tuple (see above, return_exception)
     """
     try:
         r = requests.request(
             method,
             '{}/v0.2/{}'.format(MTLS_ENDPOINT, url),
             cert=(CLIENT_CERT_PATH, CLIENT_KEY_PATH),
-            headers=get_mtls_header(debug=debug, dev=dev),
+            headers=get_mtls_header(dev=dev),
             **kwargs
         )
 
-        if (debug_on_ok or not r.ok) and debug:
-            mtls_req_error_log(url, method.upper(), requester_name, r)
+        if not requester_name:
+            requester_name = "({})".format(url)
+
+        req_error_log(method.upper(), requester_name, r, log_on_ok=log_on_ok, caller='mtls_request')
         if return_exception:
             return r, None
         else:
             return r
 
     except requests.exceptions.RequestException as e:
-        print('wott-agent :: mtls_request :: rises exception: {}'.format(e))
+        logger.exception("mtls_request :: rises exception:")
         if return_exception:
             return None, e
         else:
             return None
 
 
-def get_claim_token(debug=False, dev=False):
-    setup_endpoints(dev, debug)
+def get_claim_token(dev=False):
+    setup_endpoints(dev)
     can_read_cert()
 
-    response = mtls_request('get', 'claimed', debug=debug, dev=dev, requester_name="Get Device Claim Info")
+    response = mtls_request('get', 'claimed', dev=dev, requester_name="Get Device Claim Info")
     if response is None or not response.ok:
-        print('Did not manage to get claim info from the server.')
+        logger.error('Did not manage to get claim info from the server.')
```

Comment:
Pls pay attention that your changes in some conditions will not provide output required to `debian/postinst`

Reply:
fixed by changing the logging format in manual modes.